components/eventing-controller/api/v1alpha1/subscription_conversion.go

```diff
@@ -1,19 +1,340 @@
+			dst.Spec.Config[v1alpha2.WebhookAuthClientSecret] = src.Spec.ProtocolSettings.WebhookAuth.ClientSecret
+			dst.Spec.Config[v1alpha2.WebhookAuthTokenURL] = src.Spec.ProtocolSettings.WebhookAuth.TokenURL
+			dst.Spec.Config[v1alpha2.WebhookAuthScope] = strings.Join(src.Spec.ProtocolSettings.WebhookAuth.Scope, ",")
+		}
+	}
+}
+
+// setV1ProtocolFields converts the protocol-related fields from v1alpha1 to v1alpha2 Subscription version
+func (src *Subscription) setV1ProtocolFields(dst *v1alpha2.Subscription) {
+	if protocol, ok := dst.Spec.Config[v1alpha2.Protocol]; ok {
+		src.Spec.Protocol = protocol
+	}
 
+	if currentMode, ok := dst.Spec.Config[v1alpha2.ProtocolSettingsContentMode]; ok {
+		src.Spec.ProtocolSettings = &ProtocolSettings{}
+		src.Spec.ProtocolSettings.ContentMode = &currentMode
+	}
+	if qos, ok := dst.Spec.Config[v1alpha2.ProtocolSettingsQos]; ok {
+		src.Spec.ProtocolSettings.Qos = &qos
+	}
+	if exemptHandshake, ok := dst.Spec.Config[v1alpha2.ProtocolSettingsExemptHandshake]; ok {
+		handshake, err := strconv.ParseBool(exemptHandshake)
+		if err != nil {
+			handshake = true
+		}
+		src.Spec.ProtocolSettings.ExemptHandshake = &handshake
+	}
+	if src.Spec.ProtocolSettings != nil {
+		if authType, ok := dst.Spec.Config[v1alpha2.WebhookAuthType]; ok {
+			src.Spec.ProtocolSettings.WebhookAuth = &WebhookAuth{}
+			src.Spec.ProtocolSettings.WebhookAuth.Type = authType
+		}
+		if grantType, ok := dst.Spec.Config[v1alpha2.WebhookAuthGrantType]; ok {
+			src.Spec.ProtocolSettings.WebhookAuth.GrantType = grantType
+		}
+		if clientID, ok := dst.Spec.Config[v1alpha2.WebhookAuthClientID]; ok {
+			src.Spec.ProtocolSettings.WebhookAuth.ClientID = clientID
+		}
+		if secret, ok := dst.Spec.Config[v1alpha2.WebhookAuthClientSecret]; ok {
+			src.Spec.ProtocolSettings.WebhookAuth.ClientSecret = secret
+		}
+		if token, ok := dst.Spec.Config[v1alpha2.WebhookAuthTokenURL]; ok {
+			src.Spec.ProtocolSettings.WebhookAuth.TokenURL = token
+		}
+		if scope, ok := dst.Spec.Config[v1alpha2.WebhookAuthScope]; ok {
+			src.Spec.ProtocolSettings.WebhookAuth.Scope = strings.Split(scope, ",")
+		}
+	}
+}
+
+// setV2SpecTypes sets event types in the Subscription Spec in the v1alpha2 way
+func (src *Subscription) setV2SpecTypes(dst *v1alpha2.Subscription) error {
+	for _, filter := range src.Spec.Filter.Filters {
+		if dst.Spec.Source == "" {
+			dst.Spec.Source = filter.EventSource.Value
+		}
+		if dst.Spec.Source != "" && filter.EventSource.Value != dst.Spec.Source {
+			return errors.New(multipleSourceErrMsg)
+		}
+		dst.Spec.Types = append(dst.Spec.Types, filter.EventType.Value)
+	}
 	return nil
 }
+
+// natsSpecConfigToV2 converts the v1alpha2 Spec config to v1alpha1
+func (src *Subscription) natsSpecConfigToV1(dst *v1alpha2.Subscription) {
+	if maxInFlightMessages, ok := dst.Spec.Config[v1alpha2.MaxInFlightMessages]; ok {
+		intVal, err := strconv.Atoi(maxInFlightMessages)
+		if err == nil {
```

Comment:
What if there is an error ?

Reply:
improved this, I propagate the error now to the main convert function.